backend/api.test/Services/InspectionAreaService.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Api.Controllers.Models;
+using Api.Database.Models;
+using Api.Services;
+using Api.Test.Database;
+using Microsoft.Extensions.DependencyInjection;
+using Testcontainers.PostgreSql;
+using Xunit;
+
+namespace Api.Test.Services
+{
+    public class InspectionAreaServiceTest : IAsyncLifetime
+    {
+        public required DatabaseUtilities DatabaseUtilities;
+        public required PostgreSqlContainer Container;
+        public required IMissionRunService MissionRunService;
+        public required IInspectionAreaService InspectionAreaService;
+
+        public async Task InitializeAsync()
+        {
+            (Container, string connectionString, var connection) =
+                await TestSetupHelpers.ConfigurePostgreSqlDatabase();
+            var factory = TestSetupHelpers.ConfigureWebApplicationFactory(
+                postgreSqlConnectionString: connectionString
+            );
+            var serviceProvider = TestSetupHelpers.ConfigureServiceProvider(factory);
+
+            DatabaseUtilities = new DatabaseUtilities(
+                TestSetupHelpers.ConfigurePostgreSqlContext(connectionString)
+            );
+            MissionRunService = serviceProvider.GetRequiredService<IMissionRunService>();
+            InspectionAreaService = serviceProvider.GetRequiredService<IInspectionAreaService>();
+        }
+
+        public Task DisposeAsync() => Task.CompletedTask;
+
+        [Fact]
+        public async Task TestTasksInsidePolygon()
+        {
+            var installation = await DatabaseUtilities.NewInstallation();
+            var plant = await DatabaseUtilities.NewPlant(installation.InstallationCode);
+            var inspectionArea = await DatabaseUtilities.NewInspectionArea(
+                installation.InstallationCode,
+                plant.PlantCode
+            );
+            inspectionArea.AreaPolygonJson =
+                @"{
+                    ""zmin"": 0,
+                    ""zmax"": 10,
+                    ""positions"": [
+                        { ""x"": 0, ""y"": 0 },
+                        { ""x"": 0, ""y"": 10 },
+                        { ""x"": 10, ""y"": 10 },
+                        { ""x"": 10, ""y"": 0 }
```

Comment:
Here, we are just checking for a square polygon. Maybe sufficient for now, but an idea here could be to define a more complex polygon and use property-based testing to test it. Issue for another time probably.

Reply:
Yeah, though that was good enough for the test without having to think to much